enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,647 @@
+1. Cluster Creator applies a label to Agents if necessary so that they match the ClusterDeployment's selector.
+1. Once there are enough ready Agents of each role to fulfill the expected number as expressed on the ClusterDeployment, installation begins.
+
+#### Install Device
+
+The Agent resource Spec will include a field on which to specify the storage device
+where the OS should be installed. That field must be set by a platform-specific
+controller or some other actor that understands the underlying host.
+
+For bare metal, the BareMetalHost resource Spec already includes a
+RootDeviceHints section that will be utilized. The new Baremetal Agent
+Controller (previously described in the "Create Cluster" scenario) will use the
+BareMetalHost's RootDeviceHints and the Agent's discovery data to populate this
+field on the Agent.
+
+#### Day 2 Add Node Virtualmedia Multicluster
+
+This scenario takes place from a hub cluster, adding a worker node to a spoke
+cluster. This assumes that a ClusterDeployment exists in the hub cluster and
+has the baremetal-operator component installed (which currently requires
+setting the infrastructure platform to "baremetal").
+
+1. Infra Owner creates a BareMetalHost resource with a label that matches an InstallEnv selector.
+1. The Baremetal Agent Controller adds the discovery ISO URL to the BareMetalHost.
+1. baremetal-operator uses redfish virtualmedia to boot the live ISO on the BareMetalHost.
+1. The Agent starts up and reports back to the assisted service, which creates an Agent resource in the hub cluster. The Agent is labeled with the labels that were specified in the InstallEnv's Spec, optionally including a label to match the AgentSelector field on the ClusterDeployment.
+1. If the Agent does not get a default label matching a ClusterDeployment, a user or other automation must add a label to match the AgentSelector field on the appropriate ClusterDeployment.
+1. Cluster Creator or Infra Owner adds a reference to the ClusterDeployment onto the InstallEnv if it is not already present. This reference confirms that the ClusterDeployment, which resides in a different namespace, is authorized to consume Agents from this InstallEnv.
+1. The Agent's Role field in its spec is assigned a value if a corresponding label and value were present on its BareMetalHost. (only "worker" is supported for now on day 2)
+1. The Agent is marked as Approved via a field in its Spec based on being recognized as running on the known BareMetalHost.
+1. The Agent runs through the validation and inspection phases. The results are shown on the Agent's Status, and eventually a condition marks the Agent as "ready".
+1. The Baremetal Agent Controller adds inspection data found on the Agent's Status to the BareMetalHost.
```

Comment:
I assume this is so we get the IP address, etc.? Do we need that in the multi-cluster case, though? In a regular cluster we need the host, machine, and node to have matching IP details so the CSR for the node will be approved. What approves the CSR for nodes in this case, where the deployed cluster platform is not "baremetal" and the host resources are outside of the cluster that was deployed?

Reply:
This scenario assumes that the spoke cluster does have platform "baremetal". A BareMetalHost resource gets created on the spoke cluster in part to enable the same CSR approval workflow that is already working on that platform.

Reply:
So the CSR for the host in the spoke cluster will be approved by something in the hub cluster?

Reply:
The CSR for the host in the spoke cluster will be approved in the spoke cluster by the same component and mechanism that approves them today for the baremetal platform. Creating the BareMetalHost and a corresponding Machine with correct details in the spoke cluster should make that work. Unless there's a detail we're missing?

Reply:
There isn't anything in this section talking explicitly about data ending up in the spoke cluster, so it wasn't clear that was the case. Maybe that's all part of "installation of that host begins" on line 420?

Reply:
Yes, that's one of the things assisted does as part of its workflow. We could dig up the details if necessary, but I don't think those details are central to this proposal.